pylint/lint/pylinter.py

```diff
@@ -1326,6 +1326,8 @@ def _report_evaluation(self):
         except Exception as ex:  # pylint: disable=broad-except
             msg = f"An exception occurred while rating: {ex}"
         else:
+            if self.stats.by_msg.get("fatal", 0):
+                note = 0.0
```

Comment:
I think the code to change is in ``Pylinter.generate_reports()``, we could add typing and change the type from ``Optional[float]`` to ``float``.

Reply:
Sorry, I'm having trouble following the suggestion. Can you add more details? I may have my own solution in mind too specifically to see the suggestion.

Reply:
I think the returned score should be modified at the source where it's calculated, not modified later on. Seems like it's here that we sometime return None: https://github.com/PyCQA/pylint/blob/main/pylint/lint/pylinter.py#L1311

`` note = eval(evaluation, {}, stats_dict)  `` will raise a ``DivideByZero`` error if we do not handle the special case with 0 statement.

Reply:
When running my regression test, `self.stats.statement` is 50, not 0, so that's why I'm not sure how to get closer to the "source". If I can get closer, I will! ;)

Reply:
Sorry I thought the score was not calculated properly when there is 0 statement. What is the issue when there is a fatal and > 0 statements ?

Reply:
I see better now. We would have to change the `evaluation` expression. Fatal is not even taken into account in the score, only error:

https://github.com/PyCQA/pylint/blob/35254c29eb3a0f1c7847cfcb3451501a4180373d/pylintrc#L105